### DISCUSSION TITLE
Implementation of missing columns in DL3 Observation Index table for Eventdisplay

### DISCUSSION THREAD

**Discussion by sona-patel** [VERITAS-Observatory/V2DL3#57] [General]
- 1 to 5: Required columns
- 6 to 40: Optional columns sometime need for observation selection or data quality checks
- Status:
   - Present:  In current DL3 version
   - In progress: Yet to be tested

|  Sr. No.  |  Column Name  | Type (Unit)  |  Description    | Status  |
| ------ | ------ | ------|  ------ | ------|
|	 1 |	OBS_ID | int  | 	Unique observation identifier (Run number)|	Present |
|	2 |	RA_PNT | 	float (deg)|	Observation pointing right ascension |	Present |
|	3  |	DEC_PNT|	float (deg)|	Observation pointing declination |	Present |
|	4|	TSTART|	float (s)|	Start time of observation relative to the reference time |	Present|
|	5|	TSTOP|	float (s)|	End time of observation relative to the reference time|	Present|
|	6|	ZEN_PNT|	float (deg)|	Observation pointing zenith angle at observation mid-time|	Present|
|	7|	ALT_PNT|	float (deg)|	Observation pointing altitude at observation mid-time|	Present|
|	8|	AZ_PNT|	float (deg)|	Observation pointing azimuth at observation mid-time|	Present|
|	9|	ONTIME|	float (s)|	Total observation time (including dead time)|	Present|
|	10|	LIVETIME |	float (s)|	Total livetime (observation minus dead time)|	Present|
|	11|	DEADC | float|	Dead time correction|	Present|
|	12|	DATE-OBS|	string|	Observation start date|	Present |
|	13|	TIME-OBS|	string|	Observation start time| |
|	14|	DATE-END|	string|	Observation end date|	Present  |	
|       15|	TIME-END|	string|	Observation end time | |
|       16|	N_TELS | 	int|	Number of participating telescopes|	Present|
|	17|	TELLIST|	string|	Telescope IDs|	Present|
|	18|	QUALITY|	int|	Observation data quality. The following levels of data quality are defined (-1: unknown, 0:best, 1:medium, 2:bad) | Present (see #72) |
|	19|	OBJECT|	string|	Primary target of the observation|	Present |
|	20|	GLON_PNT|	float (deg)|	Observation pointing Galactic longitude|	In progress|
|	21|	GLAT_PNT|	float (deg)|	Observation pointing Galactic latitude|	In progress|
|	22|	RA_OBJ|	float (deg)|	Right ascension of OBJECT|	Present |
|	23|	DEC_OBJ|	float (deg)|	Declination of OBJECT|	Present |
|	24|	TMID|	float (s)|	Mid time of observation relative to the reference time| |
|	25|	TMID_STR|	string|	Mid time of observation in UTC string format: “YYYY-MM-DD HH:MM:SS” | |
|       26| EVENT_COUNT | int|	Number of events in run|	In progress|
|	27|	EVENT_RA_MEDIAN|	float (deg)|	Median right ascension of events|	In progress|
|	28|	EVENT_DEC_MEDIAN|	float (deg)|	Median declination of events|	In progress|
|	29|	EVENT_ENERGY_MEDIAN|	float (?)*|	Median energy of events ||
|       30|	EVENT_TIME_MIN|	double (s)|	First event time | |
|	31|	EVENT_TIME_MAX|	double (s)|	Last event time| |
|	32|	BKG_SCALE|	float | Observation-dependent background scaling factor | |
|	33|	TRGRATE|	float (Hz)|	Mean system trigger rate | |
|       34| 	ZTRGRATE|	float (Hz)|	Zenith equivalent mean system trigger rate | |
|	35|	MUONEFF|	float |	Mean muon efficiency | |
|	36|	BROKPIX|	float|	Fraction of broken pixels | |
|	37 |	AIRTEMP|	float (deg C)|	Mean air temperature at ground during the observation | |
|	38|	PRESSURE|	float (hPa)|	Mean air pressure at ground during the observation | |
|	39|	NSBLEVEL|	float |	Measure for NSB level | present (PR #150) |
|	40|	RELHUM|	float|	Relative humidity | |

**Comment by GernotMaier** [comment]
Thanks for this. I think in general this is ok for VERITAS.

A few comments, some of them are very special cases:

1. RA_PNT/DEC_PNT
- might vary for very few special runs during the run and for the first run of a GRB alert
- which epoch is RA/DEC? J2000 or current epoch?
- affects also ZEN_PNT and ALT_PNT
2. DEADC - what is the unit here? Say we have 15% deadtime, is this value 0.15 or 1-0.15, or? 
3. What do you write in to the TELLIST, is it T1, T2, ...
4. QUALITY - did you discuss this already? Do we have to follow the data quality assessments are can we continue to use the VERITAS standards?
5. EVENT_COUNT: is that number of all events after g/h cuts?
6. BKG_SCALE: what is that?
7. ZTRGRATE: what is that?
8. MUONEFF: why is that called muon efficiency? I understand what it does, but we have other methods to 

A lot of the other values are very HESS specific, we need to have a look if this is good enough for VERITAS or if we want to add a few others.

**Reply by sona-patel** [reply]
Thank you for pointing out these!

1. RA_PNT/DEC_PNT: 
 - Its J2000 epoch. 
 - About special runs having GRB alert: There is a separate extension called `POINTING` proposed in open gamma-astro-data-format. This extension contains RA_PNT, DEC_PNT (mandatory columns) and ALT_PNT, AZ_PNT (optional columns). But this extension table is in proposal stage, so I think it will be taken care in future. Until that time the VERITAS DL3 data producers should treat these special runs carefully.

2. DEADC: The value of 0.15 is stored for a deadtime of 15%.

3. TELLIST: It is T0, T1, T2, T3 - straight away imported from anasum ROOT file.

4. We need to discuss about `QUALITY` standards to be adopted.

5. EVENT_COUNT: Yes, at DL3 level the events are gamma-like, so all events after g/h cuts.

6. BKG_SCALE: This needs discussion to understand what it exactly corresponds to. 

7-8.  ZTRGRATE  and MUONEFF I think are HESS specific at this moment, as @GernotMaier pointed out, so we need to look into it how we should adopt these definitions in case we decide to use.

**Comment by GernotMaier** [comment]
1. might be easier to consider separating the special GRBs runs into two.

2. Note that we never talk about T0 - telescope numbers visible to humans are always T1-T4.

**Reply by sona-patel** [reply]
My mistake, `TELLIST` is not exactly imported from anasum ROOT file. This string, **T0, T1, T2, T3** is produce within V2DL3 (https://github.com/VERITAS-Observatory/V2DL3/blob/9205c299c95ef64227ea15f03f396cb7a470966e/pyV2DL3/eventdisplay/util.py#L8)

**Comment by GernotMaier** [comment]
Please use numbering starting from T1.

> On 31. May 2021, at 11:42, sonal patel ***@***.***> wrote:
> 
> 
> My mistake, TELLIST is not imported from anasum ROOT file. This string, T0, T1, T2, T3 is produce within V2DL3 (https://github.com/VERITAS-Observatory/V2DL3/blob/9205c299c95ef64227ea15f03f396cb7a470966e/pyV2DL3/eventdisplay/util.py#L8)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 



---
Dr Gernot Maier
CTA
Deutsches Elektronen-Synchrotron DESY 
Ein Forschungszentrum der Helmholtz-Gemeinschaft
Platanenallee 6, D-15738 Zeuthen, Germany

Tel.: +49  33 7627 7598
Internet: www.desy.de/cta
Besucheradresse: 1X07

**Reply by sona-patel** [reply]
Ok, pull request will be followed for `TELLIST` with required numbering. Thanks!

**Reply by sona-patel** [reply]
`TELLIST` string is **T1, T2, T3, T4** - PR #59
